hudi-common/src/main/java/org/apache/hudi/common/table/read/InputSplit.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi.common.table.read;
+
+import org.apache.hudi.common.model.FileSlice;
+import org.apache.hudi.common.model.HoodieBaseFile;
+import org.apache.hudi.common.model.HoodieLogFile;
+import org.apache.hudi.common.table.cdc.HoodieCDCUtils;
+import org.apache.hudi.common.util.Option;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class InputSplit {
```

Comment:
Javadocs?  Is this only intended to be used by the FGReader or engine integration as well (e.g., Spark has `PartitionedFile` and `HoodiePartitionFileSliceMapping` stores file slices).  Is there a plan to consolidate into using `InputSplit` for consistency in the future?

Reply:
While this is moving the class with new getters, would be good to see this should be used as the data structure for distributing Hudi splits.

Reply:
`InputSplit` can be kept in a separate class as is, since it's used across the FGReader, record buffer, and the buffer initializer.